### DISCUSSION TITLE
How I automated my Photos library backup with osXPhotos

### DISCUSSION THREAD

**Discussion by Juggler00** [RhetTbull/osxphotos#1763] [Show and tell]
I wanted to share how I have now (finally) implemented, tested and verified my weekly backup of my iPhoto library. First to note is that my Mac is set to download all photos. I am also relying on the fact that APFS, on a copy, creates a clone of the file. This means the following operations consume very little additional space, even though it appears I'm creating a full backup of my Photos library on disc. Thanks to @RhetTbull for this tip! (https://github.com/RhetTbull/osxphotos/discussions/1711)

Some quick notes:

- my Mac is set to download all photos from my iPhoto library
- I have a Linux server that I'm "rsync"ing to. This server does all my fancy off-site backups

First, I created a bash script to do the osxphotos backup, rsync the new files to my server, then notify [healthchecks.io](https://healthchecks.io) that it has completed (this way I get notified if it fails).

**osx_photos_backup.sh**
```
#!/bin/bash
/Users/USERNAME/.local/bin/osxphotos export /Users/USERNAME/Pictures/backup --update --export-by-date --skip-edited --library /Users/USERNAME/Pictures/Photos\ Library.photoslibrary --sidecar XMP --sidecar-drop-ext
/usr/bin/rsync -az /Users/USERNAME/Pictures/backup/ MYSERVER:/LOCALPATH/photos_backup/
/usr/bin/curl -m 10 --retry 5 https://hc-ping.com/XXXX
```
Next, to automate this, I created a launchd .plist file that tells my Mac when to run the above script (in this case, Fridays at 6:00am).

**com.USERNAME.photoBAckupAndArchive.plist**
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
    <dict>
        <key>Label</key>
        <string>com.USERNAME.photoBackupAndArchive</string>
        <key>Program</key>
        <string>/Users/USERNAME/Library/Scripts/osx_photos_backup.sh</string>   
        <key>StartCalendarInterval</key>
        <dict>
            <key>Weekday</key>
            <integer>5</integer>
            <key>Hour</key>
            <integer>6</integer>
            <key>Minute</key>
            <integer>0</integer>
        </dict>
    </dict>
</plist>
```

Now, I just need to activate the above...
```
 ~ % mkdir ~/Library/LaunchAgents/
 ~ % mkdir ~/Library/Scripts/
 ~ % chmod +x osx_photos_backup.sh
 ~ % cp ca.USERNAME.photoBackupAndArchive.plist ~/Library/LaunchAgents/
 ~ % cp osx_photos_backup.sh ~/Library/Scripts/
 ~ % launchctl load ~/Library/LaunchAgents/com.USERNAME.photoBackupAndArchive.plist
```

Hopefully others will find this helpful. Note that if you want to make any changes to either your script or .plist, make sure you unload the plist first (i.e. `launchctl unload NAMEOFPLIST.plist`), make the change, then reload as per above.

**Comment by RhetTbull** [comment]
Thanks for sharing! Glad osxphotos has been useful for you!
